### PR TITLE
MiniSitePreview: Give component fetch responsibility

### PR DESCRIPTION
**Pull request by spen**
This PR expands on a recent set of changes made in #26790. At the time I wasn't sure of the best way to abstract the state/fetch logic but I think I've settled on the logic changes found in this PR.
I've also tested these changes in the context of the signup importer flow and they work a charm there, too.

### To Test

- Go to `settings/import/` ([local](http://calypso.localhost:3000/settings/import/)/[calypso.live](https://calypso.live/?branch=update/give-mini-site-preview-fetch-responsibility)) and pick a site
- Choose the Wix option and enter a valid wix site (for example: https://smt593.wixsite.com/wowz/static-page )
- Tip: Set your network filter to `calypso_site_importer_site_` to better watch for relevant analytics calls.
- hit continue
  - You should see the a preview  snapshot of the given site with the expected 'browser chrome'
  - You should see an analytics event fire - make sure that the params here are as expected, particularly the `time_taken_ms` value, as the logic for that is now more complex.

### CODE REVIEW

**Comment by spen** [comment]
Thanks for taking a look @andfinally :)
Have you more details on the `onFetchSuccess` being called twice? I didn't see this myself while testing.

**Comment by andfinally** [comment]
I've just tried again in a fresh Incognito browser, and I'm still getting the two `calypso_site_importer_site_preview_success` requests. The first happens when the component displays the preview, which is as you'd expect. The second happens during the import process, after I've clicked the `Yes! Start Import` button.

These are the two requests:

importer-success-1

importer-success-2

And here's a gif of what I'm doing:

importer-success

**Comment by spen** [comment]
Thanks for the details! I'll investigate :) It sounds like it's an issue with the component being re-mounted somewhere in the process.

File: client/components/mini-site-preview/index.jsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ import { invoke } from 'lodash';
  */
 import { loadmShotsPreview } from 'my-sites/importer/site-importer/site-preview-actions';
 
-export class MiniSitePreviewWrapper extends Component {
+export class MiniSitePreview extends Component {
```
Comment (andfinally) [inline client/components/mini-site-preview/index.jsx]: I'm not sure why we `export` here as well as at the bottom, maybe that's just my ignorance.

File: client/my-sites/importer/site-importer/site-importer-input-pane.jsx [hunk 2]
```diff
@@ -411,18 +417,19 @@ class SiteImporterInputPane extends React.Component {
...
-							importData={ this.state.importData }
-							isLoading={ this.state.loading }
-							resetImport={ this.resetImport }
-							startImport={ this.importSite }
-							site={ this.props.site }
-						/>
-					</div>
-				) }
+				{ this.state.importStage === 'importable' &&
+					! ( importStarted && ! this.state.loading ) && (
```
Comment (andfinally) [inline client/my-sites/importer/site-importer/site-importer-input-pane.jsx]: The negation's a bit hard to follow - would it work to say `( ! importStarted && this.state.loading )`?

**Comment by andfinally** [comment]
Er sorry Spen, everything's looking good on the code side, and the analytics calls are firing nicely, but I'm getting quite a long interval where I'm looking at a blank chrome placeholder before the site preview loads. I get this on calypso.live and on local when I try to import a site the screenshot API hasn't captured before.

Currently, prod seems to hide the whole import preview pane behind a spinner till the screenshot's available. The way it is here may give users the feeling something's not worked properly. This gif shows what I'm seeing on local in real time - it's a bit quicker on calypso.live, but still an odd experience, IMHO.

fluffy

File: client/components/mini-site-preview/index.jsx (hunk 1 above)
Comment (spen) [reply client/components/mini-site-preview/index.jsx]: It's good practice to export the base component... but even better practice to test the component, which I've not done here 😬

**Comment by spen** [comment]
Regarding the mShots preview being shown without an image to start with, instead showing a placeholder while the image loads - I'd say that this is expected behaviour, sort of. 

If we want the component itself to handle it's own loading in a contained manner, we need it to show a loading state. We could communicate this up to the parent in the `onFetchSuccess` callback but I'm not keen on doing that.

Instead, we could just use redux state, but that breaks us away from being a self-contained component... we'd be ditching this PR and having each implementation of the dumb component handle the fetching and receiving actions. I'm not totally against this really.

**Comment by marekhrabe** [comment]
Can't we just render the preview component in the importing UI sooner so it starts fetching in parallel with the other API request?

**Comment by spen** [comment]
> Can't we just render the preview component in the importing UI sooner 

We could mount it as soon as we have the url, we just need to make sure we're only mounting it once. Either way there is a time in between mounting (and fetching) and receiving the preview image where, of course, there's nothing to display but a placeholder.

We could just `display: none` on the component but that could leave us guessing when is best to show the component.

I'm at a point where I feel this is an incredible waste of time though. It's already taken a chunk of time to get this to where it is and it's not really all that important.

I'm closing this for that reason. If anybody else fancies giving this a go, feel free to.
